generators/profilegen/src/resources/templates/module_template.template

```diff
@@ -22,42 +26,43 @@ module Azure::<%- if !@individual_gem_profile -%>Profiles::<%- end -%><%- if @in
     # <%= @class_name %>
     #
     class <%= @class_name %>Class
-      attr_reader <% @operation_types.each do |operation_type| %>:<%= operation_type[:operation_name_ruby] %>, <% end %>:configurable, :base_url, :options, :model_classes
+      attr_reader <%- @clients_ops_mapper.each do |key, value| -%><% value.each do |operation_type| %><%- if(operation_type[:operation_name_ruby] != 'DO_NOT_ADD') -%>:<%= operation_type[:operation_name_ruby] %>, <% end %><% end %><% end %>:configurable, :base_url, :options, :model_classes
 
       def initialize(configurable, base_url=nil, options=nil)
         @configurable, @base_url, @options = configurable, base_url, options
-        client = <%= @management_client %>.new(configurable.credentials, base_url, options)
-        if(client.respond_to?(:subscription_id))
-          client.subscription_id = configurable.subscription_id
+
+        <%- @clients_ops_mapper.each_with_index do |(key, value), index| -%>
+        client_<%= index %> = <%= key %>.new(configurable.credentials, base_url, options)
+        if(client_<%= index %>.respond_to?(:subscription_id))
+          client_<%= index %>.subscription_id = configurable.subscription_id
         end
-        <%- @operation_types.each do |operation_type| -%>
-        @<%= operation_type[:operation_name_ruby] %> = client.<%= operation_type[:operation_name_ruby] %>
+        <%- value.each do |operation_type| -%>
+        <%- if(operation_type[:operation_name_ruby] != 'DO_NOT_ADD') -%>
+        @<%= operation_type[:operation_name_ruby] %> = client_<%= index %>.<%= operation_type[:operation_name_ruby] %>
+        <%- end -%>
+        <%- end -%>
+
         <%- end -%>
         @model_classes = ModelClasses.new
       end
 
-      <%- if @individual_gem_profile -%>
       #
       # Method to get the client object
       #
       # @return Client object
       #
+      <%- if @individual_gem_profile && @versions_clients_mapper.keys.length == 0 -%>
       def get_client
         client = <%= @versions_clients_mapper[@versions_clients_mapper.keys[0]] %>.new(@configurable.credentials, @base_url, @options)
         client.subscription_id = configurable.subscription_id
         return client
       end
       <%- else -%>
-      #
-      # Method to get the client object based on the version
-      # If the version is invalid, an exception is raised
-      #
-      # @param version [String] The version of the required
-      # client object.
-      #
-      # @return Client object
-      #
+      <%- if @individual_gem_profile -%>
+      def get_client(version = '<%= @versions_clients_mapper.keys[@versions_clients_mapper.keys.length - 1] %>')
+      <%- else -%>
       def get_client(version = '<%= @default_rp_client_version %>')
```

Comment:
where's the default_rp_client_version version getting picked up from? Should default_rp_client_version be assigned "@versions_clients_mapper.keys[@versions_clients_mapper.keys.length - 1]" for the case of individual gem profile, and so the same template code is applicable?